pkg/scheduler/dependency_graph.go

```diff
@@ -83,6 +92,35 @@ func newScheduledResourceFor(r interfaces.Resource, context *GraphContext) *Sche
 	}
 }
 
+type DependencyList client.DependencyList
+
+func (d *DependencyList) Len() int {
+	return len(d.Items)
+}
+
+func (d *DependencyList) Less(i, j int) bool {
+	if d.Items[i].CreationTimestamp.Equal(d.Items[j].CreationTimestamp) {
+		return d.Items[i].UID < d.Items[j].UID
+	}
+	return d.Items[i].CreationTimestamp.Before(d.Items[j].CreationTimestamp)
+}
+
+func (d *DependencyList) Swap(i, j int) {
```

Comment:
Why add these methods? Are there any new interfaces that this list needs to implement? If yes, please do so by adding static check for interface implementation.